src/Serialization.cpp

```diff
@@ -0,0 +1,1310 @@
+    auto memoize_eviction_key_serialized = serialize_expr(builder, memoize_eviction_key);
+    return Halide::Serialize::CreateFuncSchedule(builder, store_level_serialized, compute_level_serialized, builder.CreateVector(storage_dims_serialized), builder.CreateVector(bounds_serialized),
+                                                 builder.CreateVector(estimates_serialized), builder.CreateVector(wrappers_serialized), memory_type, memoized, async, memoize_eviction_key_serialized.first, memoize_eviction_key_serialized.second);
+}
+
+flatbuffers::Offset<Halide::Serialize::Specialization> Serializer::serialize_specialization(flatbuffers::FlatBufferBuilder &builder, const Specialization &specialization) {
+    auto condition_serialized = serialize_expr(builder, specialization.condition);
+    auto definition_serialized = serialize_definition(builder, specialization.definition);
+    auto failure_message_serialized = serialize_string(builder, specialization.failure_message);
+    return Halide::Serialize::CreateSpecialization(builder, condition_serialized.first, condition_serialized.second, definition_serialized, failure_message_serialized);
+}
+
+flatbuffers::Offset<Halide::Serialize::Definition> Serializer::serialize_definition(flatbuffers::FlatBufferBuilder &builder, const Definition &definition) {
+    auto is_init = definition.is_init();
+    auto predicate_serialized = serialize_expr(builder, definition.predicate());
+    std::vector<uint8_t> values_types;
+    values_types.reserve(definition.values().size());
+    std::vector<flatbuffers::Offset<void>> values_serialized;
+    values_serialized.reserve(definition.values().size());
+    for (const auto &value : definition.values()) {
+        auto value_serialized = serialize_expr(builder, value);
+        values_types.push_back(value_serialized.first);
+        values_serialized.push_back(value_serialized.second);
+    }
+    std::vector<uint8_t> args_types;
+    args_types.reserve(definition.args().size());
+    std::vector<flatbuffers::Offset<void>> args_serialized;
+    args_serialized.reserve(definition.args().size());
+    for (const auto &arg : definition.args()) {
+        auto arg_serialized = serialize_expr(builder, arg);
+        args_types.push_back(arg_serialized.first);
+        args_serialized.push_back(arg_serialized.second);
+    }
+    auto stage_schedule_serialized = serialize_stage_schedule(builder, definition.schedule());
+    std::vector<flatbuffers::Offset<Halide::Serialize::Specialization>> specializations_serialized;
+    for (const auto &specialization : definition.specializations()) {
+        specializations_serialized.push_back(serialize_specialization(builder, specialization));
+    }
+    auto source_location_serialized = serialize_string(builder, definition.source_location());
+    return Halide::Serialize::CreateDefinition(builder, is_init, predicate_serialized.first, predicate_serialized.second, builder.CreateVector(values_types),
```

Comment:
Nit here and elsewhere: we don't have an explicit line-width limit in Halide style guidelines but these are getting wide enough that I have to make the GitHub diff window fullscreen on my 4k monitor to avoid line wrapping so, uh, maybe some of these could be made a bit less wide :-)